internal/api/impl/auth/token.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"net/http"
+
+	"github.com/perses/perses/internal/api/shared"
+	"github.com/perses/perses/internal/api/shared/crypto"
+	"github.com/sirupsen/logrus"
+)
+
+type tokenManagement struct {
+	jwt crypto.JWT
+}
+
+func (tm *tokenManagement) accessToken(login string, provider string) (string, []*http.Cookie, error) {
+	accessToken, err := tm.jwt.SignedAccessToken(login, provider)
+	if err != nil {
+		logrus.WithError(err).Errorf("unable to generate the access token")
+		return "", nil, shared.InternalError
+	}
+	jwtHeaderPayloadCookie, signatureCookie := tm.jwt.CreateAccessTokenCookie(accessToken)
+
+	return accessToken, []*http.Cookie{jwtHeaderPayloadCookie, signatureCookie}, nil
```

Comment:
based on the usage you are doing with these methods, probably you can keep the set of the Cookies in this method. It would avoid to do it everytime we are contacting these methods

Reply:
The cookie set is not always the same. The OIDC library force me to use the http writer, in the classic echo endpoints, we use the ctx.

Reply:
I introduced a setCookie function parameter to save some lines of code